viewing_party/party.py

```diff
@@ -1,23 +1,154 @@
+import copy
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if not title or not genre or not rating:
+        return None
+    new_movie = {
+            "title": title,
+            "genre": genre,
+            "rating": rating
+            }
+    return new_movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+
+
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+
+    return user_data
+
+
+def watch_movie(user_data, title):
+    new_watchlist = []
+    new_watched = []
+
+    for movie in user_data["watchlist"]:
+        new_watchlist.append(movie)
+    
+    for movie in user_data["watched"]:
+        new_watched.append(movie)
+
+    for movie in new_watchlist:
+        if movie["title"] == title:
+            new_watched.append(movie)  
+            new_watchlist.remove(movie)
+            break
+
+    updated_user_data = {
+        "watchlist": new_watchlist,
+        "watched": new_watched
+    }
+    
+    return updated_user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    if not len(user_data["watched"]):
+        return 0.0
+    sum = 0
+    for movie in user_data["watched"]:
+        sum += movie["rating"]
+
+    average = sum / len(user_data["watched"])
+    return average
+
+def get_most_watched_genre(user_data):
+    if not user_data:
+        return None
+
+    genres = {}
+    for movie in user_data["watched"]:
+        movie_genre = movie["genre"]
+        count = genres.get(movie_genre, 0)
+        genres[movie_genre] = count + 1
 
+    entry = None
+    max_number_of_occurences = 0
+
+    for genre, count in genres.items():
+        if count > max_number_of_occurences:
+            entry = genre
+            max_number_of_occurences = count
+
+    return entry
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_unique_watched(user_data):

```

Comment:
The `get_unique_watched` function should return a list of movies that only a user has watched but their friends have not watched. Currently, your function has two responsibilities: first it creates a list of friends movies and then it identifies which movies only the user has watched. 

Consider putting the logic on lines 86-91 in a helper function like `get_friends_movies` and calling that helper function here. Doing so would make this method more concise. 

Since this method is concerned with getting unique movies watched, we don't need to have the implementation of getting a list of movies watched by friends exposed in this method here.